hal/src/nRF52840/ble_hal.cpp

```diff
@@ -1034,6 +1034,10 @@ int BleObject::Broadcaster::setAdvertisingParams(const hal_ble_adv_params_t* par
         tempParams.inc_tx_power = false;
     } else {
         memcpy(&tempParams, params, std::min(tempParams.size, params->size));
+        if (tempParams.primary_phy != BLE_PHYS_1MBPS && tempParams.primary_phy != BLE_PHYS_CODED) {
```

Comment:
`BLE_PHYS_AUTO` is allowed and the SoftDevice will apply `BLE_PHYS_1MBPS` implicitly. For old user application, the `primary_phy` field is presented as the reserved field, which is initialized to 0 (now corresponding to `BLE_PHYS_AUTO`), the new Device OS should be backward compatible and accept the `BLE_PHYS_AUTO`.

And we should remove the `reserved` field in `hal_ble_adv_params_t` to make sure the struct is word aligned still. And we should bump the `BLE_API_VERSION` in ble_hal.h, since we have changed the struct filed.

Reply:
Good catch! Added the check for `BLE_PHYS_AUTO`. Thought about checking for `BLE_PHYS_AUTO` and changing to `BLE_PHYS_1MBPS`, but if the SoftDevice already does it, we don't need to.

I removed the `reserved` field and bumped `BLE_API_VERSION`